server/src/endpoints/common.rs

```diff
@@ -133,15 +136,16 @@ impl ResponseError for BadStoreRequest {
 /// If store_event receives a non empty store_body it will use it as the body of the
 /// event otherwise it will try to create a store_body from the request.
 ///
-pub fn handle_store_like_request<F, R>(
+pub fn handle_store_like_request<F, R, I>(
     meta: EventMeta,
     start_time: StartTime,
     request: HttpRequest<ServiceState>,
     extract_envelope: F,
     create_response: R,
 ) -> ResponseFuture<HttpResponse, BadStoreRequest>
 where
-    F: FnOnce(Bytes, EventMeta) -> Result<Envelope, BadStoreRequest> + 'static,
+    F: FnOnce(&HttpRequest<ServiceState>, EventMeta, usize) -> I + 'static,
```

Comment:
Let's not pass the `max_event_payload_size` parameter in here. Instead, each endpoint can have its own logic to determine the max size. As a matter of fact, the Minidump endpoint has different limits than other endpoints.

Reply:
I propose to pass in the entire config object.

Reply:
No need, config is available in every endpoint already.